misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java

```diff
@@ -269,23 +273,47 @@ public static boolean addTypeDefinition(TypeDefinitionNode typeDefinition, Modul
             }
             module.errors.add(new Error(typeName, getDocFromMetadata(metaDataNode), isDeprecated(metaDataNode), type));
         } else if (typeDefinition.typeDescriptor().kind() == SyntaxKind.TUPLE_TYPE_DESC) {
-            module.types.add(getTupleTypeModel((TupleTypeDescriptorNode) typeDefinition.typeDescriptor(),
+            module.tupleTypes.add(getTupleTypeModel((TupleTypeDescriptorNode) typeDefinition.typeDescriptor(),
+                    typeName, metaDataNode, semanticModel, module));
+        } else if (typeDefinition.typeDescriptor().kind() == SyntaxKind.TABLE_TYPE_DESC) {
```

Comment:
Can we use a `switch` statement for these instead of using several `if-else` blocks?

Reply:
Otherwise can factor out `typeDefinition.typeDescriptor().kind()` to a variable

Reply:
We cannot use `switch` statement since there are some `else if` conditions which does not simply equal to `syntaxkind`. we can create a variable for `typeDefinition.typeDescriptor().kind()`